File: Backend/Dixen.API/Dixen.Repo/Model/Entities/Hall.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Text.Json.Serialization;
using System.Threading.Tasks;

namespace Dixen.Repo.Model.Entities
{
    public class Hall
    {
        public int Id { get; set; }
        public string Name { get; set; } = string.Empty;
        public int Capacity { get; set; }
        public int EventId { get; set; }
        public Evnt Event { get; set; }
        //public List<EventAttendance> Attendances { get; set; } = new();
        public int VenueId { get; set; }
        public Venue Venue { get; set; }
    }
}

File: Frontend/DixenUI/src/app/Models/hall.ts
// import { Venue } from "./venue";

// export class Hall {
//   id: number = 0;
//   name: string = "";
//   capacity: number = 0;
//   eventId: number = 0;
//   // attendanceCount?: number;
//   venueId: number = 0;
// }
import { Evnt } from "./Evnt";
import { Venue } from "./venue";

export class Hall {
  id: number = 0;
  name: string = "";
  capacity: number = 0;
  eventId: number = 0;
  event?: Evnt;        
  venueId: number = 0;
  venue?: Venue;  
  seatsAvailable: number = 0;     
  isDeleted: boolean = false;
}
